File: src/main/web/src/types/metric.ts
/**
 * Metric 类型定义
 */
export interface Metric {
  id?: string;
  metricType: string;
  metricName: string;
  metricValue: number;

  runId?: string;
  agentId?: string;
  traceId?: string;
  spanId?: string;

  labels?: Record<string, any>;

  timestamp?: string;
  createdAt?: string;
}

/**
 * Metric 类型枚举
 */
export enum MetricType {
  LATENCY_NS = 'LATENCY_NS',
  THROUGHPUT = 'THROUGHPUT',
  INPUT_TOKENS = 'INPUT_TOKENS',
  OUTPUT_TOKENS = 'OUTPUT_TOKENS',
  TOTAL_TOKENS = 'TOTAL_TOKENS',
  CPU_USAGE = 'CPU_USAGE',
  MEMORY_USAGE = 'MEMORY_USAGE',
  ERROR_COUNT = 'ERROR_COUNT',
}

File: src/main/java/com/agenthub/domain/model/monitor/Metric.java
package com.agenthub.domain.model.monitor;

import lombok.Data;

import java.time.Instant;
import java.util.Map;

import static com.agenthub.common.utils.RandomUtils.randomId;

/**
 * Metric 领域模型.
 * 表示监控指标数据.
 */
@Data
public class Metric {
    private String id;
    private String metricType;
    private String metricName;

    private Double metricValue;

    private String runId;
    private String agentId;
    private String traceId;
    private String spanId;

    private Map<String, Object> labels;

    private Instant timestamp;

    private String tenantId;
    private String workspaceId;
    private Instant createdAt;

    public Metric() {
        this.id = randomId();
        this.timestamp = Instant.now();
        this.createdAt = Instant.now();
    }

    public static Metric create(String metricType, String metricName, Double value) {
        Metric metric = new Metric();
        metric.metricType = metricType;
        metric.metricName = metricName;
        metric.metricValue = value;
        return metric;
    }
}
